pkg/controller/backup_storage_location_controller.go

```diff
@@ -169,21 +173,38 @@ func (r *BackupStorageLocationReconciler) logReconciledPhase(defaultFound bool,
 
 	if numUnavailable+numUnknown == len(locationList.Items) { // no available BSL
 		if len(errs) > 0 {
-			log.Errorf("Current backup storage locations available/unavailable/unknown: %v/%v/%v, %s)", numAvailable, numUnavailable, numUnknown, strings.Join(errs, "; "))
+			log.Errorf("Current BackupStorageLocations available/unavailable/unknown: %v/%v/%v, %s)", numAvailable, numUnavailable, numUnknown, strings.Join(errs, "; "))
 		} else {
-			log.Errorf("Current backup storage locations available/unavailable/unknown: %v/%v/%v)", numAvailable, numUnavailable, numUnknown)
+			log.Errorf("Current BackupStorageLocations available/unavailable/unknown: %v/%v/%v)", numAvailable, numUnavailable, numUnknown)
 		}
 	} else if numUnavailable > 0 { // some but not all BSL unavailable
-		log.Warnf("Unavailable backup storage locations detected: available/unavailable/unknown: %v/%v/%v, %s)", numAvailable, numUnavailable, numUnknown, strings.Join(errs, "; "))
+		log.Warnf("Unavailable BackupStorageLocations detected: available/unavailable/unknown: %v/%v/%v, %s)", numAvailable, numUnavailable, numUnknown, strings.Join(errs, "; "))
 	}
 
 	if !defaultFound {
-		log.Warn("There is no existing backup storage location set as default. Please see `velero backup-location -h` for options.")
+		log.Warn("There is no existing BackupStorageLocation set as default. Please see `velero backup-location -h` for options.")
 	}
 }
 
 func (r *BackupStorageLocationReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	g := kube.NewPeriodicalEnqueueSource(r.Log, mgr.GetClient(), &velerov1api.BackupStorageLocationList{}, backupStorageLocationSyncPeriod)
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&velerov1api.BackupStorageLocation{}).
+		// Ignore all events of BSL. Reconciler is supposed to run periodically.
```

Comment:
Keeping these events is better. It makes sure the status of BSL is updated immediately no matter what the sync period users configure is

Reply:
``` go
	if !storage.IsReadyToValidate(location.Spec.ValidationFrequency, location.Status.LastValidationTime, r.DefaultBackupLocationInfo.ServerValidationFrequency, log) {
		log.Debug("Validation not required, skipping...")
		return ctrl.Result{}, nil
	}
```
BSL controller has a check to only validate BSL at an 60s interval, most of the event will be skipped by this check.

Reply:
Add creation event handling for validate new creating BSL immediately.